src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisService.java

```diff
@@ -87,32 +87,34 @@ public List<String> getProviders() {
     }
 
     public Long getFactorsCount(SensitivityFactorsIdsByGroup factorIds, UUID networkUuid, String variantId, Boolean isInjectionsSet) {
-        Long containersAttributesCount = 1L;
-        if (Boolean.TRUE.equals(isInjectionsSet)) {
-            containersAttributesCount *= factorIds.getIds().get(INJECTIONS).size();
+        long containersAttributesCount = 1L;
+        Map<String, List<UUID>> ids = factorIds.getIds();
+        if (Boolean.TRUE.equals(isInjectionsSet) && ids.containsKey(INJECTIONS) && !ids.get(INJECTIONS).isEmpty()) {
+            containersAttributesCount = filterService.getIdentifiablesCountForACategory(ids.get(INJECTIONS), INJECTIONS, networkUuid, null)
+                .values()
+                .stream()
+                .reduce(Long::sum)
+                .orElse(1L);
             factorIds.getIds().remove(INJECTIONS);
         }
-        containersAttributesCount *= getFactorsCount(factorIds, networkUuid, variantId);
+        containersAttributesCount *= getBranchesAndContingenciesFactorsCount(factorIds, networkUuid, variantId);
```

Comment:
we have to change the function name ?

Reply:
no but I thought it would be better to differenciate the two methods 
I can revert